plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsApplicationResource.java

```diff
@@ -41,7 +68,7 @@ public InputStream getInputStream() throws IOException {
 
     @Override
     public long getLastModified() throws IOException {
-        File file = new File(url.getPath());
+        File file = new File(super.getLocalePath());
```

Comment:
We already have passed it to super and we don't need compute it once again. My added unit tests show it seems work.

Reply:
The problem is `super.getLocalePath()` (from the Tiles code) _doesn't guarantee return of the exact path string passed to super_.  Unfortunately the unit test can't easily check all possibilities, and changing the parameter passed risks breaking existing applications unexpectedly (in hard to predict ways).

If we want to avoid re-computing the path value, then _maybe we should just add a new attribute_ to cache the URL path string in this PR and modify it like this:
```java
    final String urlPath;
    // Code up to the constructor
    public StrutsApplicationResource(URL url) throws MalformedURLException {
        urlPath = getExistedPath(url);
        super(urlPath);
        this.url = url;
    }
    // Intervening code
    @Override
    public long getLastModified() throws IOException {
        File file = new File(urlPath);
    // Rest of code
```

Reply:
fixed :) - learnt from tiles URLApplicationResource.